In [1]:
import pandas as pd
import numpy as np

In [4]:
plot_geom = pd.read_csv('/Users/ellapalacios/Downloads/tree_proj/data/WA_PLOTGEOM.csv')
plot = pd.read_csv('/Users/ellapalacios/Downloads/tree_proj/data/WA_PLOT.csv')

In [29]:
plot_geom = plot_geom.drop(columns=['CREATED_DATE', 'MODIFIED_DATE'])

In [30]:
plot_geom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20308 entries, 0 to 20307
Data columns (total 14 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CN           20308 non-null  int64  
 1   INVYR        20308 non-null  float64
 2   UNITCD       20308 non-null  float64
 3   COUNTYCD     20308 non-null  float64
 4   PLOT         20308 non-null  float64
 5   LAT          20308 non-null  float64
 6   LON          20308 non-null  float64
 7   ECOSUBCD     20308 non-null  object 
 8   HUC          20306 non-null  float64
 9   EMAP_HEX     20308 non-null  float64
 10  FVS_VARIANT  20298 non-null  object 
 11  FVS_LOC_CD   20298 non-null  float64
 12  FVS_REGION   20298 non-null  float64
 13  FVS_FOREST   20298 non-null  float64
dtypes: float64(11), int64(1), object(2)
memory usage: 2.2+ MB


In [31]:
plot_geom.sort_values(by = ['PLOT'])

CN   INVYR  UNITCD  COUNTYCD     PLOT        LAT  \
18387  714229426290487  2020.0     5.0      61.0   9741.0  47.887899   
19225  714229438290487  2020.0     9.0      65.0  13103.0  48.967794   
19173  714229428290487  2020.0     9.0      51.0  35610.0  48.261460   
18691  714229442290487  2020.0     8.0       7.0  43456.0  48.170244   
5808    22954698010497  2007.0     9.0      19.0  50002.0  48.009997   
...                ...     ...     ...       ...      ...        ...   
15757  412221778489998  2017.0     8.0      39.0  99995.0  45.886524   
4111    22399248010497  2006.0     5.0      57.0  99997.0  48.574878   
14269  346864417489998  2016.0     5.0      57.0  99997.0  48.574878   
4720     8647102010901  2006.0     8.0      77.0  99999.0  46.418657   
14883  345933933489998  2016.0     8.0      77.0  99999.0  46.418657   

              LON ECOSUBCD         HUC  EMAP_HEX FVS_VARIANT  FVS_LOC_CD  \
18387 -121.466592   M242Db  17110009.0   26074.0          WC       613.0   
19225 -117.956265   M333Ae  17020001.0   24755.0          EC       621.0   
19173 -117.097459   M333Aa  17010216.0   24379.0          IE       113.0   
18691 -120.566502   M242Dg  17020009.0   25726.0          EC       617.0   
5808  -118.506956   M333Ae  17020001.0   25007.0          EC       608.0   
...           ...      ...         ...       ...         ...         ...   
15757 -120.872634    342Ig  17070106.0   25967.0          EC       613.0   
4111  -121.556779   M242Da  17110005.0   26071.0          WC       605.0   
14269 -121.556779   M242Da  17110005.0   26071.0          WC       605.0   
4720  -119.941069    342Ig  17030003.0   25616.0          EC       617.0   
14883 -119.941069    342Ig  17030003.0   25616.0          EC       617.0   

       FVS_REGION  FVS_FOREST  
18387         6.0        13.0  
19225         6.0        21.0  
19173         1.0        13.0  
18691         6.0        17.0  
5808          6.0         8.0  
...           ...         ...  
15757         6.0        13.0  
4111          6.0         5.0  
14269         6.0         5.0  
4720          6.0        17.0  
14883         6.0        17.0  

[20308 rows x 14 columns]

We can see from above that rows 4720 and 14883 are the same plot # and location, the measurements/samples are just taken 10 years apart. Let's try dropping duplicates in PLOT# and maybe plot it and see what happens?? 



In [32]:
plot_geom_copy = plot_geom.copy()


In [40]:
single_lon = plot_geom_copy.drop_duplicates(subset=['LON'])
single_lon.sort_values(by = ['PLOT'])

CN   INVYR  UNITCD  COUNTYCD     PLOT        LAT  \
18387  714229426290487  2020.0     5.0      61.0   9741.0  47.887899   
19225  714229438290487  2020.0     9.0      65.0  13103.0  48.967794   
19173  714229428290487  2020.0     9.0      51.0  35610.0  48.261460   
18691  714229442290487  2020.0     8.0       7.0  43456.0  48.170244   
5808    22954698010497  2007.0     9.0      19.0  50002.0  48.009997   
...                ...     ...     ...       ...      ...        ...   
472     24123058010900  2002.0     8.0       7.0  99993.0  47.801631   
5297    22956187010497  2007.0     6.0      31.0  99994.0  47.678942   
5595    22955998010497  2007.0     8.0      39.0  99995.0  45.886524   
4111    22399248010497  2006.0     5.0      57.0  99997.0  48.574878   
4720     8647102010901  2006.0     8.0      77.0  99999.0  46.418657   

              LON ECOSUBCD         HUC  EMAP_HEX FVS_VARIANT  FVS_LOC_CD  \
18387 -121.466592   M242Db  17110009.0   26074.0          WC       613.0   
19225 -117.956265   M333Ae  17020001.0   24755.0          EC       621.0   
19173 -117.097459   M333Aa  17010216.0   24379.0          IE       113.0   
18691 -120.566502   M242Dg  17020009.0   25726.0          EC       617.0   
5808  -118.506956   M333Ae  17020001.0   25007.0          EC       608.0   
...           ...      ...         ...       ...         ...         ...   
472   -120.984707   M242Db  17020011.0   25959.0          EC       617.0   
5297  -123.305667   M242Ai  17110018.0   26740.0          PN       609.0   
5595  -120.872634    342Ig  17070106.0   25967.0          EC       613.0   
4111  -121.556779   M242Da  17110005.0   26071.0          WC       605.0   
4720  -119.941069    342Ig  17030003.0   25616.0          EC       617.0   

       FVS_REGION  FVS_FOREST  
18387         6.0        13.0  
19225         6.0        21.0  
19173         1.0        13.0  
18691         6.0        17.0  
5808          6.0         8.0  
...           ...         ...  
472           6.0        17.0  
5297          6.0         9.0  
5595          6.0        13.0  
4111          6.0         5.0  
4720          6.0        17.0  

[10212 rows x 14 columns]

In [41]:
import plotly.express as px
import matplotlib.pyplot as plt

fig = px.scatter_geo(single_lon, lat='LAT', lon='LON',projection="mercator")

fig.update_traces(marker=dict(size=1))

fig.update_geos(
    center={"lat": 47.5, "lon": -120.0},  # Center around Washington State
    projection_scale=40  # Higher value zooms in more
)

fig.show()

In [44]:
#try dropping duplicate plot numbers

no_duplicate_plot_nums = plot_geom_copy.drop_duplicates(subset = ['PLOT'])
no_duplicate_plot_nums.sort_values(by = ['PLOT'])

CN   INVYR  UNITCD  COUNTYCD     PLOT        LAT  \
18387  714229426290487  2020.0     5.0      61.0   9741.0  47.887899   
19225  714229438290487  2020.0     9.0      65.0  13103.0  48.967794   
19173  714229428290487  2020.0     9.0      51.0  35610.0  48.261460   
18691  714229442290487  2020.0     8.0       7.0  43456.0  48.170244   
5808    22954698010497  2007.0     9.0      19.0  50002.0  48.009997   
...                ...     ...     ...       ...      ...        ...   
472     24123058010900  2002.0     8.0       7.0  99993.0  47.801631   
5297    22956187010497  2007.0     6.0      31.0  99994.0  47.678942   
5595    22955998010497  2007.0     8.0      39.0  99995.0  45.886524   
4111    22399248010497  2006.0     5.0      57.0  99997.0  48.574878   
4720     8647102010901  2006.0     8.0      77.0  99999.0  46.418657   

              LON ECOSUBCD         HUC  EMAP_HEX FVS_VARIANT  FVS_LOC_CD  \
18387 -121.466592   M242Db  17110009.0   26074.0          WC       613.0   
19225 -117.956265   M333Ae  17020001.0   24755.0          EC       621.0   
19173 -117.097459   M333Aa  17010216.0   24379.0          IE       113.0   
18691 -120.566502   M242Dg  17020009.0   25726.0          EC       617.0   
5808  -118.506956   M333Ae  17020001.0   25007.0          EC       608.0   
...           ...      ...         ...       ...         ...         ...   
472   -120.984707   M242Db  17020011.0   25959.0          EC       617.0   
5297  -123.305667   M242Ai  17110018.0   26740.0          PN       609.0   
5595  -120.872634    342Ig  17070106.0   25967.0          EC       613.0   
4111  -121.556779   M242Da  17110005.0   26071.0          WC       605.0   
4720  -119.941069    342Ig  17030003.0   25616.0          EC       617.0   

       FVS_REGION  FVS_FOREST  
18387         6.0        13.0  
19225         6.0        21.0  
19173         1.0        13.0  
18691         6.0        17.0  
5808          6.0         8.0  
...           ...         ...  
472           6.0        17.0  
5297          6.0         9.0  
5595          6.0        13.0  
4111          6.0         5.0  
4720          6.0        17.0  

[10219 rows x 14 columns]

In [45]:
fig = px.scatter_geo(no_duplicate_plot_nums, lat='LAT', lon='LON',projection="mercator")

fig.update_traces(marker=dict(size=1))

fig.update_geos(
    center={"lat": 47.5, "lon": -120.0},  # Center around Washington State
    projection_scale=40  # Higher value zooms in more
)

fig.show()

In [48]:
print('Plot #97140, comes from County # 61:       ', plot_geom.loc[18387])

Plot #97140, comes from County # 61:        CN             714229426290487
INVYR                   2020.0
UNITCD                     5.0
COUNTYCD                  61.0
PLOT                    9741.0
LAT                  47.887899
LON                -121.466592
ECOSUBCD                M242Db
HUC                 17110009.0
EMAP_HEX               26074.0
FVS_VARIANT                 WC
FVS_LOC_CD               613.0
FVS_REGION                 6.0
FVS_FOREST                13.0
Name: 18387, dtype: object


In [56]:
plot.sort_values(by = 'PLOT')
plot.loc[plot['PLOT'] == 9741]

CN           SRV_CN    CTY_CN  PREV_PLT_CN  INVYR  \
15391  714229426290487  643410130126144  75010497          NaN   2020   

       STATECD  UNITCD  COUNTYCD  PLOT  PLOT_STATUS_CD  ...  PAC_ISLAND_PNWRS  \
15391       53       5        61  9741               1  ...               NaN   

       PLOT_SEASON_NERS  PRECIPITATION  PREV_MICROPLOT_LOC_RMRS  \
15391               NaN            NaN                      NaN   

       PREV_PLOT_STATUS_CD_RMRS  REUSECD1  REUSECD2  REUSECD3  \
15391                       NaN       NaN       NaN       NaN   

       GRND_LYR_SAMPLING_STATUS_CD  GRND_LYR_SAMPLING_METHOD_CD  
15391                          NaN                          NaN  

[1 rows x 69 columns]

The CN # from WA_PLOTGEOM is 714229426290487 and the CN # from WA_PLOT is 714229426290487.. which are the same. Let's figure out where the plot is, likely using a scatter plot, but I have to think on this one. If it's in what we know to be a forested region, THEN we might be able to make some cross-refrences. 